test/unit-tests/vast_test.js

```diff
@@ -2327,4 +2329,103 @@ describe('ad_manager_vast', function() {
     expect(trackingUrlsPinged.closeUrl).to.be                   (1);
   });
 
+  it('VAST: Wrapper ads should be properly parsed into the adTrackingInfo object', function() {
+    var embed_code = "embed_code";
+    var vast_ad = {
+      type: "vast",
+      first_shown: 0,
+      frequency: 2,
+      ad_set_code: "ad_set_code",
+      time:10,
+      position_type:"t",
+      url:"1.jpg"
+    };
+    var content = {
+      embed_code: embed_code,
+      ads: [vast_ad]
+    };
+    vastAdManager.initialize(amc);
+    vastAdManager.loadMetadata({"html5_ssl_ad_server":"https://blah",
+      "html5_ad_server": "http://blah"}, {}, content);
+    initialPlay();
+    vastAdManager.initialPlay();
+    
+    // Wrapper ads could be visualized as a tree with parents and children,
+    // but in this case, it looks more like a linked list:
+    // wrapper-parent-1 -> wrapper-parent-2 -> 6654644 (Inline Linear Ad)
+    var parentDepthOneId = "wrapper-parent-1";
+    var parentDepthTwoId = "wrapper-parent-2";
+    var leafId = "6654644"; // Ad ID from linearXML file
+
+    // need to fake wrapper ajax calls
+    vastAdManager.onVastResponse(vast_ad, wrapper1XML);
+    vastAdManager.onVastResponse(vast_ad, wrapper2XML, parentDepthOneId);
+    vastAdManager.onVastResponse(vast_ad, linearXML, parentDepthTwoId);
+   
+    var adTrackingInfo = vastAdManager.adTrackingInfo;
+
+    // adTrackingInfo should have the three ads parsed
+    expect(_.keys(adTrackingInfo).length).to.be(3);
+    expect(_.has(adTrackingInfo, parentDepthOneId)).to.be(true);
+    expect(_.has(adTrackingInfo, parentDepthTwoId)).to.be(true);
+    expect(_.has(adTrackingInfo, leafId)).to.be(true);
+
+    var parentDepthOneObject = adTrackingInfo[parentDepthOneId];
+    var parentDepthTwoObject = adTrackingInfo[parentDepthTwoId];
+    var leafObject = adTrackingInfo[leafId];
+
+    // Ad Tracking Objects should have correct wrapper parent IDs
+    expect(parentDepthOneObject.wrapperParentId).to.be(null);
+    expect(parentDepthTwoObject.wrapperParentId).to.be(parentDepthOneId);
+    expect(leafObject.wrapperParentId).to.be(parentDepthTwoId);
+
+    expect(parentDepthOneObject.adObject).to.not.be(null);
```

Comment:
You are checking that the object exists. Can you check and see if some of the messages exist within the right object?